flatm8s/src/App.js

```diff
@@ -1,32 +1,113 @@
-import React from 'react'
-import SignUp from './SignUp'
-import Home from './Home'
-import Error from './Error'
-import UserLogin from "./UserLogin"
-import ConnectToFlat from "./ConnectToFlat";
-import { Switch, Link, Route } from 'react-router-dom'
+import React from 'react';
+import SignUp from './pages/SignUp';
+import Home from './pages/Home';
+import Error from './pages/Error';
+import UserLogin from "./pages/UserLogin";
+import ConnectToFlat from "./pages/ConnectToFlat";
+import {Switch, Link, Route, useLocation} from 'react-router-dom'
+import {Button, Toolbar, Typography} from "@material-ui/core";
+import {makeStyles} from "@material-ui/core/styles";
 
 
 function App() {
+    const location = useLocation()
   return (
-      <>
-          <nav>
-              <ul>
-                  <li><Link to='/'>Home</Link></li>
-                  <li><Link to='/SignUp'>Sign Up</Link></li>
-                  <li><Link to='/UserLogin'>Login</Link></li>
-                  <li><Link to='/ConnectFlat'>Find a flat</Link></li>
-              </ul>
-          </nav>
-          <Switch>
-              <Route path="/" component={Home} exact />
-              <Route path="/SignUp" component={SignUp} />
-              <Route path="/UserLogin" component={UserLogin} />
-              <Route path="/ConnectFlat" component={ConnectToFlat} />
-              <Route component={Error} />
-          </Switch>
-      </>
+          <NavBar props={location.state}/>
   );
 }
 
-export default App;
+const useStyles = makeStyles(() => ({
+    root: {
+        backgroundColor: '#3f51b5',
+        color: 'white',
+        flexGrow: 1,
+        borderRadius: 5,
+    },
+    title: {
+        flexGrow: 1,
+    },
+    colour: {
+        color: "white",
+    },
+}));
+
+function NavBar(props) {
+    const classes = useStyles()
+
+    function signOut() {
+        console.log('this works', props?.props)
+        let myHeaders = new Headers();
+        myHeaders.append("Content-Type", "application/json");
+        myHeaders.append("X-Authorization", props.props.token);
+
+        let requestOptions = {
+            method: 'POST',
+            headers: myHeaders,
+            redirect: 'follow'
+        };
+        fetch("http://localhost:4000/users/logout", requestOptions)
+            .then(response => response.text())
+            .then(result => {
+                console.log(result)
+
+            })
+            .catch(error => console.log('error', error));
+    }
+
+    let test
```

Comment:
I imagine you don't want this anymore, it is still showing in the centre of the nav

Reply:
Im just leaving it there so i have a clear indicator while im testing